### DISCUSSION TITLE
Errors in kubernetes yaml after upgrading to spark 3.5.0

### DISCUSSION THREAD

**Discussion by paulpaul1076** [stackabletech/spark-k8s-operator#325] [Q&A] [answered]
After we updated spark operator to 3.5.0, we now get this error:

Error from server (BadRequest): error when creating "C:\\Users\\wnwnn\\IdeaProjects\\firestarter\\firestarter\\src\\main\\resources\\kubernetes\\app-action-event-batch.yml": SparkApplication in version "v1alpha1" cannot be handled as a SparkApplication: strict decoding error: unknown field "spec.driver.resources", unknown field "spec.driver.volumeMounts", unknown field "spec.executor.instances", unknown field "spec.executor.resources", unknown field "spec.executor.volumeMounts", unknown field "spec.sparkImagePullPolicy"

Did something change? How do you specify these things after the upgrade?

**Comment by adwk67** [comment, marked as answer]
Thanks for the feedback @paulpaul1076! There were a number of breaking changes in the 23.11 release and we have endeavoured to flag these up and to document how to redefine the resource yamls in the release notes. For the spark-k8s-operator you can find this [here](https://docs.stackable.tech/home/stable/release-notes#_stackable_operator_for_apache_spark). In a nutshell you will need to move the configuration elements (volumes, logging, resources etc.) down one level to be underneath "config", using the new product image structure, and renaming the executor `instances` to `replicas`.

**Reply by adwk67** [reply]
And....apologies for not consistently carrying this through to the operator docs: I see that [here](https://docs.stackable.tech/home/stable/spark-k8s/crd-reference) the config level has not yet been included. I'll open an issue for that.

**Reply by adwk67** [reply]
Issue opened [here](https://github.com/stackabletech/spark-k8s-operator/issues/326).
